StRoot/StFwdUtils/StFwdAnalysisMaker.cxx

```diff
@@ -1,78 +1,110 @@
 #include "StFwdUtils/StFwdAnalysisMaker.h"
-#include "StFwdTrackMaker/Common.h"
 
-#include <limits>
-#include <map>
-#include <string>
-#include <string>
-#include <vector>
+#include "TVector3.h"
 
 #include "StEvent/StEvent.h"
-#include "StEvent/StGlobalTrack.h"
-#include "StEvent/StHelixModel.h"
-#include "StEvent/StPrimaryTrack.h"
-#include "StEvent/StRnDHit.h"
-#include "StEvent/StRnDHitCollection.h"
-#include "StEvent/StTrack.h"
-#include "StEvent/StTrackGeometry.h"
-#include "StEvent/StTrackNode.h"
-#include "StEvent/StPrimaryVertex.h"
 #include "StEvent/StEnumerations.h"
-#include "StEvent/StTrackDetectorInfo.h"
-#include "StEvent/StFttPoint.h"
-#include "StEvent/StFcsHit.h"
 #include "StEvent/StFcsCluster.h"
 #include "StEvent/StFttCollection.h"
 #include "StEvent/StFcsCollection.h"
-#include "StEvent/StTriggerData.h"
-#include "StEvent/StFstHitCollection.h"
-#include "StEvent/StFstHit.h"
 #include "StEvent/StFwdTrackCollection.h"
-#include "StChain/StChainOpt.h"
-
-#include "StEventUtilities/StEventHelper.h"
 
 #include "StMuDSTMaker/COMMON/StMuDstMaker.h"
 #include "StMuDSTMaker/COMMON/StMuDst.h"
 #include "StMuDSTMaker/COMMON/StMuFwdTrack.h"
+#include "StMuDSTMaker/COMMON/StMuFwdTrackCollection.h"
 
+#include "St_base/StMessMgr.h"
 
-#include "tables/St_g2t_fts_hit_Table.h"
-#include "tables/St_g2t_track_Table.h"
-#include "tables/St_g2t_vertex_Table.h"
-#include "tables/St_g2t_event_Table.h"
+#include "StEvent/StFwdTrack.h"
+#include "StFcsDbMaker/StFcsDb.h"
 
-#include "StarMagField/StarMagField.h"
+//________________________________________________________________________
+StFwdAnalysisMaker::StFwdAnalysisMaker() : StMaker("fwdAna"){
+    setLocalOutputFile( "" ); // default off
+};
+int StFwdAnalysisMaker::Finish() { 
+    
+    if ( mLocalOutputFile != "" ){
+        auto prevDir = gDirectory;
+        
+        // output file name
+        TFile *fOutput = new TFile(mLocalOutputFile, "RECREATE");
+        fOutput->cd();
+        for (auto nh : mHists) {
+            nh.second->SetDirectory(gDirectory);
+            nh.second->Write();
+        }
 
-#include "St_base/StMessMgr.h"
-#include "StarClassLibrary/StPhysicalHelix.hh"
-#include "StarClassLibrary/SystemOfUnits.h"
+        // restore previous directory
+        gDirectory = prevDir;
 
+        LOG_INFO << "Done writing StFwdAnalysisMaker output to local file : " << mLocalOutputFile << endm;
+    }
```

Comment:
Someone please remind me whether it is okay for analysis makers to write histograms in their own root files or they should add them to the common *_hists.root file.

Reply:
My interpretation of this code is that as long as `mLocalOutputFile` isn't set (via `setLocalOutputFile()`), the histograms do go to the common hist.root file.

Reply:
Sounds good.